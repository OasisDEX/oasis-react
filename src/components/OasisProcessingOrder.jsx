import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";
// import ImmutablePropTypes from 'react-immutable-proptypes';

import styles from "./OasisProcessingOrder.scss";
import OasisTransactionStatusWrapper from "../containers/OasisTransactionStatus";

const propTypes = PropTypes && {
  localStatus: PropTypes.string,
  txType: PropTypes.string,
  txTimestamp: PropTypes.number
};
const defaultProps = {};



export class OasisProcessingOrder extends PureComponent {
  static customBlock() {
    return (
      <div>Process order</div>
    );
  }
  transactionInfo() {
    const { localStatus, txType, txTimestamp } = this.props;
    return (
      <OasisTransactionStatusWrapper
        customBlock={OasisProcessingOrder.customBlock()}
        txType={txType}
        localStatus={localStatus}
        txTimestamp={txTimestamp}
      />
    );
  }
  render() {
    return <div className={styles.base}>{this.transactionInfo()}</div>;
  }
}

OasisProcessingOrder.displayName = "OasisProcessingOrder";
OasisProcessingOrder.propTypes = propTypes;
OasisProcessingOrder.defaultProps = defaultProps;
export default OasisProcessingOrder;
