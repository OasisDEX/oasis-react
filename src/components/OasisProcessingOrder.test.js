/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisProcessingOrder from './OasisProcessingOrder';


describe('(Component) OasisProcessingOrder', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <OasisProcessingOrder {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
