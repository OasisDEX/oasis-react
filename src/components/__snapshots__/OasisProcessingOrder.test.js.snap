// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisProcessingOrder should render 1`] = `
<div>
  <Connect(OasisTransactionStatus)
    customBlock={
      <div>
        Process order
      </div>
    }
  />
</div>
`;
